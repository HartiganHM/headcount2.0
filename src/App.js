import React, { Component } from 'react';
import DistrictRepository from './helper'
import schoolData from '../data/kindergartners_in_full_day_program.js'
import './App.css';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: {}
    }
  }

  componentDidMount() {
    let data = new DistrictRepository(schoolData);
    this.setState({data})
  }

  render() {
    return (
      <div>Welcome To Headcount 2.0</div>
    );
  }
}

export default App;
