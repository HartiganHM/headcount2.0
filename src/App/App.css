.app {
    padding: 1.2rem;
}