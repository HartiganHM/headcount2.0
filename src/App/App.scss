@import '../variables.scss';

.app {
    display: flex;
    flex-direction: column;
    font-family: $main-font;

    .header {
        background-color: $accent-color;
        color: $white;
        font-family: $accent-font;
        font-size: 6.4rem;
        padding: 1.2rem;
        text-align: center;
    }
}