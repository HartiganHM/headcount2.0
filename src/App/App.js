import React, { Component } from 'react';
import DistrictRepository from '../helper'
import schoolData from '../../data/kindergartners_in_full_day_program.js'
import './App.css';

import Search from '../Search/Search';
import CardContainer from '../CardContainer/CardContainer';

class App extends Component {
  constructor() {
    super();
    this.state = {
      data: {}
    }

    this.filterData = this.filterData.bind(this);
  }

  filterData(input) {
    let newData = this.state.data.findAllMatches(input);

    console.log(newData)
  }

  componentDidMount() {
    let data = new DistrictRepository(schoolData);
    this.setState({data})
  }

  render() {
    return (
      <div className='app'>
        <span className='header'>Headcount 2.0</span>
        <Search filterCards={this.filterData}/>
        <CardContainer data={this.state.data} />
        <footer className='footer'>Developed by Matt Renn and Hugh Hartigan</footer>
      </div>
    );
  }
}

export default App;