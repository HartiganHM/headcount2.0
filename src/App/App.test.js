import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount, render } from 'enzyme';
import importedData from '../../data/kindergartners_in_full_day_program.js';

describe('App Tests', () => {
  let renderedApp;

  beforeEach(() => {
    renderedApp = shallow(<App />);
  });

  it('App should exist', () => {
    expect(renderedApp).toBeDefined();
  });

  it('App should match the snapshot', () => {
    expect(renderedApp).toMatchSnapshot();
  });

  it('Should have default state', () => {
    const defaultState = {};

    renderedApp.state().data = defaultState;
    expect(renderedApp.state().data).toEqual(defaultState);
  });

  it('Should take data as state', () => {
    const defaultState = {};
    const expectedState = importedData;

    renderedApp.state().data = defaultState;
    expect(renderedApp.state().data).toEqual(defaultState);

    renderedApp.instance().setState({ data: expectedState });
    expect(renderedApp.state().data).toEqual(expectedState);
  });
});
