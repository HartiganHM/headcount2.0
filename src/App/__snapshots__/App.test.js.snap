exports[`App Tests App should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span>
          Headcount 2.0
</span>,
        <Search />,
        <CardContainer />,
        <footer>
          Developed by Matt Renn and Hugh Hartigan
</footer>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Headcount 2.0",
        },
        "ref": null,
        "rendered": "Headcount 2.0",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Developed by Matt Renn and Hugh Hartigan",
        },
        "ref": null,
        "rendered": "Developed by Matt Renn and Hugh Hartigan",
        "type": "footer",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span>
            Headcount 2.0
</span>,
          <Search />,
          <CardContainer />,
          <footer>
            Developed by Matt Renn and Hugh Hartigan
</footer>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Headcount 2.0",
          },
          "ref": null,
          "rendered": "Headcount 2.0",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Developed by Matt Renn and Hugh Hartigan",
          },
          "ref": null,
          "rendered": "Developed by Matt Renn and Hugh Hartigan",
          "type": "footer",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
